apps/studio/src/server/modules/collection/collection.router.ts

```diff
@@ -127,6 +136,44 @@ export const collectionRouter = router({
             by: user,
           })
 
+          const indexJson = createCollectionIndexJson(collection.title)
+
+          const blob = await tx
+            .insertInto("Blob")
+            .values({ content: jsonb(indexJson) })
+            .returning("Blob.id")
+            .executeTakeFirstOrThrow()
+
+          const addedResource = await tx
+            .insertInto("Resource")
+            .values({
+              title: collection.title,
+              permalink: INDEX_PAGE_PERMALINK,
+              siteId,
+              parentId: collection.id,
+              draftBlobId: blob.id,
+              type: ResourceType.IndexPage,
+              state: ResourceState.Draft,
+            })
+            .returningAll()
+            .executeTakeFirstOrThrow()
+            .catch((err) => {
+              if (get(err, "code") === PG_ERROR_CODES.uniqueViolation) {
+                throw new TRPCError({
+                  code: "CONFLICT",
+                  message: "A resource with the same permalink already exists",
+                })
+              }
+              throw err
+            })
```

Comment:
we need to auto create collection index pagse going forward because this is relied on by our new collection tags feature (and by extension, collection categories)